src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/NuGetCredentials.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Credentials;
+using NuGet.Protocol;
+using NuGet.Protocol.Plugins;
+
+namespace Microsoft.DotNet.UpgradeAssistant.MSBuild
+{
+    public static class NuGetCredentials
+    {
+        public static void ConfigureCredentialService(bool interactive, ILogger logger)
+        {
+            if (HttpHandlerResourceV3.CredentialService == null)
+            {
+                HttpHandlerResourceV3.CredentialService = new Lazy<ICredentialService>(
+                    () => new CredentialService(
+                        providers: new AsyncLazy<IEnumerable<ICredentialProvider>>(async () => await GetCredentialProvidersAsync(logger).ConfigureAwait(false)),
+                        nonInteractive: !interactive,
+                        handlesDefaultCredentials: PreviewFeatureSettings.DefaultCredentialsAfterCredentialProviders));
+            }
+        }
+
+        public static async Task<IEnumerable<ICredentialProvider>> GetCredentialProvidersAsync(ILogger logger)
```

Comment:
if user input never makes it to the credentials provider then does it matter if the input is interactive? Seems we should hardcode that to always nonInteractive or wire-up credentials and userInput

Reply:
The caller (`PackageLoader`) is using `UserInput` to determine what to pass for this. Considering this comment and Taylor's suggestion to get NuGetCredentials from DI, it may be better to just have this class be a DI-retrieved service that has the UserInput dependency directly.